driver/src/price_estimation/dexag/mod.rs

```diff
@@ -57,13 +81,26 @@ where
     Api: DexagApi + Sync,
 {
     fn get_prices(&self, tokens: &[Token]) -> Result<HashMap<TokenId, u128>> {
+        if tokens.is_empty() {
+            return Ok(HashMap::new());
+        }
+
+        self.initialize_if_needed()
```

Comment:
Can this method return the api token list as part of the result? This way we don't have to borrow and unwrap.

Reply:
Agree that it's nicer. Implemented.